src/frontend/xgboost.cc

```diff
@@ -44,12 +43,21 @@ namespace {
 
 typedef float bst_float;
 
+template <typename T>
+inline T* BeginPtr(std::vector<T>& vec) {  // NOLINT(*)
+  if (vec.empty()) {
```

Comment:
Can you use the data member function?

Reply:
Done. All uses of `BeginPtr` in `frontend/xgboost.cc` involve non-empty containers, so we can just use `data()` instead.